woocommerce/google-listings-and-ads
# Integration with new WC Navigation

### Changes proposed in this Pull Request:

Closes #285 .

This PR makes our GLA plugin work with the new WC Navigation experience.

#### Technical notes

In `js/src/index.js`, `title` seemed to be not used at all, I removed them in https://github.com/woocommerce/google-listings-and-ads/pull/605/commits/3b577f52f90ffe5939a020d50145eb08e22d0518.

The WC Navigation menu item text depends on breadcrumbs (see code: https://github.com/woocommerce/woocommerce-admin/blob/7d70ebdd168eb91ccf0c6be4c7aec1ef7f07189d/client/layout/navigation.js#L68). Without changing the breadcrumbs, all the menu items show "Google Listings & Ads" (this was one of the major pains for me to figure out in this PR). So I had to fix the breadcrumbs, which was done in https://github.com/woocommerce/google-listings-and-ads/pull/605/commits/706c8d8aa7c8efa761c52c7d8c0d5eb1c1f5bd29.

In the WC Navigation menu, when you click on menu items at the same hierarchy level, there will be menu transition animation, which does not match normal user experience. I believe this is an issue with WC Navigation. I looked into WC Payment https://github.com/Automattic/woocommerce-payments and their plugin is having the same issue as well.

https://user-images.githubusercontent.com/417342/118877482-15754b80-b921-11eb-90a7-b9dfe0cca276.mov

WC Navigation menu items are displayed based on `path` in the URL. If the `path` does not correspond to any items in the WC Navigation, then it will show the root WooCommerce level. This presents a problem for us in the report pages, because we have Programs Report at `path=/google/reports/programs` and Products Report at `path=/google/reports/products`, but we only have one "Reports" menu item in the WC Navigation, which I have tied to `path=/google/reports/programs`. This means that when we are displaying Programs Report, WC Navigation will display the menu item correctly, but when we navigate to Products Report, WC Navigation will show the root WooCommerce level. You can see this in the demo video below. To solve this problem, I'd suggest two ways:

1. Make the "Reports" a menu category, clicking on it will display another WC Navigation level with the two reports. This means we would need to ditch the Reports tab-based navigation. One problem with this during my experimentation is that the "Reports" menu category would appear at the very top above "Dashboard". I'm guessing there is some sort of logic to put menu category first followed by menu items.
2. Instead of using `path=/google/reports/programs` and `path=/google/reports/products`, we use another parameter to represent the type of report, e.g. `path=/google/reports&report=programs` and `path=/google/reports&report=products`. With this, the `path` always be `path=/google/reports`, and the WC Navigation menu item "Reports" will always show correctly. If this is ok, we can work on this in a separate issue.

Update 2021-05-25: I've created issue https://github.com/woocommerce/google-listings-and-ads/issues/665 to address the URL `path` issue.

### Screenshots:

Demo video with my voice: https://d.pr/v/jLpVI2

https://user-images.githubusercontent.com/417342/118874469-89adf000-b91d-11eb-950d-b2a45def6f22.mp4

### Detailed test instructions:

Note: You can turn on/off the WC Navigation experience at https://gla1.test/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features.

1. Turn on the WC Navigation experience.
2. When Setup MC is not completed yet, there should be only one Google Listings and Ads menu item, and clicking on it will display the Get Started page.
3. Click the Set up free listings button to start Setup MC flow. The Setup MC flow pages should display OK. (This uses the `FullScreen` component.)
4. When Setup MC is completed, the Google Listings and Ads menu would become a category. Clicking on it will show the next level of menu items.
5. Browse through the Dashboard, Reports, Product Feed, and Settings pages. They should display OK.
6. Go to Dashboard and click on the Edit button for Free Listings. The Edit Free Listings page should display OK. (This uses the `FullContainer` component.)
7. Turn off the WC Navigation experience and repeat the above steps.

### Changelog Note:

Integrate GLA plugin with the new WC Navigation experience.
